src/core/vertex.js

```diff
@@ -596,40 +597,65 @@ p5.prototype.quadraticVertex = function(cx, cy, x3, y3) {
  * vertex(30, 75);
  * endShape();
  * </code>
+ * <code>
+ * var vectorArray = [
+ *  new p5.Vector(30, 20),
+ *  new p5.Vector(85, 20),
+ *  new p5.Vector(85, 75),
+ *  new p5.Vector(30, 75)
+ * ]
+ * beginShape(POINTS);
+ * for (var i = 0; i < vectorArray.length; i++) {
+ *  vertex(vectorArray[i]);
+ * }
+ * endShape();
+ * </code>
  * </div>
  *
  * @alt
  * 4 black points in a square shape in middle-right of canvas.
  *
  */
 p5.prototype.vertex = function(x, y, moveTo) {
-  var args = new Array(arguments.length);
-  for (var i = 0; i < args.length; ++i) {
-    args[i] = arguments[i];
-  }
```

Comment:
I believe this loop is created so that we can have an actual array of arguments (with associated array functions) instead of an array-like object. See [here](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments). Whether it is needed or not, I can't really say although it seem all the code beneath are not using `args`, I can't really say whether it will be needed in the future, is it worth keeping for now?